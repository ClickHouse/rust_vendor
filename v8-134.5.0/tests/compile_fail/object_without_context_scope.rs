// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let mut scope = v8::HandleScope::new(&mut isolate);
  let _object = v8::Object::new(&mut scope);
}

fn mock<T>() -> T {
  unimplemented!()
}
