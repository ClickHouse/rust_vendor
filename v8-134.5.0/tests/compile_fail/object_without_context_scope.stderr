error[E0308]: mismatched types
 --> tests/compile_fail/object_without_context_scope.rs:6:33
  |
6 |   let _object = v8::Object::new(&mut scope);
  |                 --------------- ^^^^^^^^^^ expected `&mut HandleScope<'_>`, found `&mut HandleScope<'_, ()>`
  |                 |
  |                 arguments to this function are incorrect
  |
  = note: expected mutable reference `&mut HandleScope<'_, v8::Context>`
             found mutable reference `&mut HandleScope<'_, ()>`
note: associated function defined here
 --> src/object.rs
  |
  |   pub fn new<'s>(scope: &mut HandleScope<'s>) -> Local<'s, Object> {
  |          ^^^
